### PR TITLE
Chart changes

### PR DESCRIPTION
**Pull request by chipzoller**
Changes I've made in my local fork including fixes for OpenShift support.

### CODE REVIEW

File: charts/kubecost-cloud-agent/values.yaml [hunk 1]
```diff
@@ -669,12 +702,13 @@ prometheus:
...
-    enabled: true
+    enabled: false
   # kubecost emits pre-2.0 KSM metrics, KSM is enabled by default here for backwards compatibility, but can be disabled to save resources without concern to kubecost metrics
   kubeStateMetrics:
     enabled: true
   kube-state-metrics:
-    disabled: false
+    disabled: true
+    enabled: false
   pushgateway:
```
Comment (mbolt35) [inline charts/kubecost-cloud-agent/values.yaml:712]: `kube-state-metrics` too? What's the short story here.

File: charts/kubecost-cloud-agent/values.yaml [hunk 2]
```diff
@@ -613,17 +646,17 @@ prometheus:
   podSecurityPolicy:
     enabled: false
   extraScrapeConfigs: |
-    - job_name: kubecost
+    - job_name: kubecost-cloud-agent
```
Comment (mbolt35) [inline charts/kubecost-cloud-agent/values.yaml:649]: Will upgrading here cause any issues with renaming this?

File: charts/kubecost-cloud-agent/values.yaml [hunk 3]
```diff
@@ -1,4 +1,33 @@
...
+imageVersion: lunar-sandwich.v0.1.2
+
+# Sets the destination where the Kubecost agent connects.
+cloudReportingServer: collector.app.kubecost.com:31357
+
+# Cloud Agent enables specific features designed to enhance the metrics exporter deployment
+# with enhancements designed for Kubecost Cloud 
+cloudAgent: true 
+cloudAgentKey: ""
+cloudAgentClusterId: my-cluster
```
Comment (mbolt35) [inline charts/kubecost-cloud-agent/values.yaml:14]: This is the only default I think is potentially problematic, as we want to required this be set externally. Is the best practice here to have the chart error if the variable is unset?

File: charts/kubecost-cloud-agent/values.yaml [hunk 4]
```diff
@@ -613,17 +646,17 @@ prometheus:
...
       scrape_timeout: 60s
       metrics_path: /metrics
       scheme: http
       dns_sd_configs:
       - names:
-        - {{ template "cost-analyzer.serviceName" . }}
+        - {{ .Release.Name }}-cloud-agent
         type: 'A'
-        port: 9003
+        port: 9005
```
Comment (mbolt35) [inline charts/kubecost-cloud-agent/values.yaml:659]: Why the port change?

Comment (mbolt35) [reply charts/kubecost-cloud-agent/values.yaml:659]: Oh, I see - you moved it from the values!

File: charts/kubecost-cloud-agent/values.yaml (hunk 1 above)
Comment (mbolt35) [reply charts/kubecost-cloud-agent/values.yaml:712]: I also noticed you moved this from the values.

File: charts/kubecost-cloud-agent/values.yaml [hunk 5]
```diff
@@ -370,7 +401,9 @@ kubecostMetrics:
   # from the cost-model. If the deployment is not enabled, the metrics will continue to be emitted from
   # the cost-model.
   exporter:
-    enabled: false
+    enabled: true
```
[the remark this answers is not in the document]
Comment (mbolt35) [reply charts/kubecost-cloud-agent/values.yaml:404]: Hah - also copied from values. Ignore :)

**Comment by mbolt35** [comment]
I think the only question I have here is what is the best practice for pointing the agent at our staging environment? Just override the values directly?

**Comment by chipzoller** [comment]
> I think the only question I have here is what is the best practice for pointing the agent at our staging environment? Just override the values directly?

Yes, staging values should come from another source and should override the default values for any internal processes including, but not limited to, CI and QA-related deployments.

File: charts/kubecost-cloud-agent/values.yaml (hunk 3 above)
Comment (chipzoller) [reply charts/kubecost-cloud-agent/values.yaml:14]: Yes, if this is a required value and must be dynamic in nature, the value should be set to empty string in the parent values file with a validation check to ensure it is not empty upon deployment.

**Comment by mbolt35** [comment]
@chipzoller What is your opinion about dropping helm altogether for our install process? Our installation requires a specific key, and I do not see any advantage to publishing any sort of of versioned helm chart for consumption outside of the specific install steps provided via the cloud service. Helm is supposed to be a useful tool, but if it requires constant maintenance and sticking to strict release standards, then we've just create a whole lot of work for resources we don't have. I'm seeing way more maintenance costs than I am value from helm.

**Comment by chipzoller** [comment]
What alternative do you suggest?

**Comment by chipzoller** [comment]
> > 1. Image name should be unique, not reuse "cost-model" even if from the same lineage/codebase.
> 
> I don't think the helm chart conventions should dictate anything about the binary. Are you suggesting we move this back into the cost-analyzer-helm-chart?

They don't, this comment isn't about the chart but about the container image. The container image repository currently used for the KC cloud agent is called `cost-model`. The reasoning for that is clear, but typically (and for many reasons) a new product such as this receives a container image which matches. That doesn't mean starting with a new codebase but distinguishing between product offerings in a clear way. For example, in this instance I would expect to see something like `kubecost-agent` or `kc-cloud-agent` or something to that effect. The other reasoning is so you can properly start with your own versioning lifecycle without conflicting with versions/tags adopted by the other. See my second comment for that.

> 
> > 4. The chart should use standard Helm release processes and tools. Release artifacts (index.yaml, tarballs) should not be in the same branch as the chart files.
> 
> Reference to this process?
> 

For example, [here](https://helm.sh/docs/howto/chart_releaser_action/), Helm's official documentation referencing their chart releaser action which is almost always employed when the source lives on GitHub, even for monorepo structures.

> Everything else I agree with, but the question remains whether or not long term maintenance of a chart is the right move for the cloud-agent.

**Pull request closed without merging**